lib/matplotlib/cbook/__init__.py

```diff
@@ -1322,18 +1327,42 @@ def _reshape_2D(X, name):
 
     *name* is used to generate the error message for invalid inputs.
     """
-    # Iterate over columns for ndarrays, over rows otherwise.
-    X = np.atleast_1d(X.T if isinstance(X, np.ndarray) else np.asarray(X))
+    # Iterate over columns for ndarrays.
+    if isinstance(X, np.ndarray):
+        X = X.T
+
+        if len(X) == 0:
+            return [[]]
+        elif X.ndim == 1 and np.ndim(X[0]) == 0:
+            # 1D array of scalars: directly return it.
+            return [X]
+        elif X.ndim in [1, 2]:
+            # 2D array, or 1D array of iterables: flatten them first.
+            return [np.reshape(x, -1) for x in X]
+        else:
+            raise ValueError("{} must have 2 or fewer dimensions".format(name))
+
+    # Iterate over rows for non-ndarrays.
     if len(X) == 0:
         return [[]]
-    elif X.ndim == 1 and np.ndim(X[0]) == 0:
+
+    result = []
+    is_1d = True
+    for xi in X:
+        xi = np.asarray(xi)
```

Comment:
Is there a reason not to use `np.asanyarray()`?

Reply:
Probably fine, but it's because the original did `np.asarray`.